### ISSUE TITLE
sync_batchnorm with gradient accumulation 

### ISSUE THREAD

**Issue by YoniChechik** [Lightning-AI/lightning#17446]
### Description & Motivation

There is an easy way to do gradient accumulatation on lighting, but as I understand the batch norm is problematic since it's calculated every forward pass.
We should fix this by enabling batch norm sync on gradient accumulatation (mainly with one gpu).

cc @borda

**Comment by awaelchli** [comment]
@YoniChechik Do you have a proposal how to do this properly? And what do you mean with "mainly on one GPU". Are you referring to the synchronization that happens in the sync batch norm layers?

**Comment by YoniChechik** [comment]
The main purpuse for using gradient accumulatation is because one don't have enough space on the gpu for weights + large batch size. This problem accours mainly when you only have one gpu...

The issue with batch-norm is it changes every forward pass, and I was wondering if this could be coupled with the sync-batch-norm to fix this
